static/js/message_edit.js

```diff
@@ -185,6 +186,44 @@ export function get_deletability(message) {
     return false;
 }
 
+export function stream_and_topic_exist_in_edit_history(message, stream_id, topic) {
+    /*  Checks to see if a stream_id and a topic match any historical
+        stream_id and topic state in the message's edit history.
+
+        Does not check the message's current stream_id and topic for
+        a match to the stream_id and topic parameters.
+     */
+    const narrow_dict = {stream_id, topic};
+    const message_dict = {stream_id: message.stream_id, topic: message.topic};
+
+    if (!message.edit_history) {
+        return false;
+    }
+
+    for (const edit_history_event of message.edit_history) {
+        if (!edit_history_event.prev_stream && !edit_history_event.prev_topic) {
+            continue;
+        }
+        if (
+            edit_history_event.prev_stream &&
+            edit_history_event.stream === message_dict.stream_id
```

Comment:
So, just to make sure I'm thinking this through clearly -- I think we expect this condition to always be true as an assertion about the correctness of the edit history data structure? It might be better to have this be done as an assertion statement rather than and `&&` condition in this loop, for readability reasons.

Reply:
Yes, you're correct. It is basically an assertion of the correctness of the edit history data structure.

Reply:
I did a quick check through the front end code for other assertion statements, but I didn't find any examples to work from and I'm not sure outside of a testing framework how I would rewrite it as such.

If we're pretty confident about the edit history data structure, would it make sense to remove this assertion?

Reply:
Yeah, let's remove it; I don't have a better plan.